src/rust/engine/engine_cffi/src/lib.rs

```diff
@@ -966,6 +966,27 @@ pub extern "C" fn run_local_interactive_process(
           Some(TempDir::new().map_err(|err| format!("Error creating tempdir: {}", err))?)
         };
 
+        let input_files_value = externs::project_ignoring_type(&value, "input_files");
+        let digest: Digest = nodes::lift_digest(&input_files_value)?;
+        if digest != EMPTY_DIGEST {
+          if run_in_workspace {
+            return Err("When running an interactive process in the workspace, use the Workspace type to materialize files.".to_string());
```

Comment:
Great!

Reply:
Rather than errorring here, could you error in `__post__init__` validation for this datatype?